src/transformers/modeling_utils.py

```diff
@@ -1347,13 +1350,17 @@ def load(module: nn.Module, prefix=""):
     def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=False):
         module_keys = set([".".join(key.split(".")[:-1]) for key in names])
 
+        # torch.nn.ParameterList is a special case where two parameter keywords
+        # are appended to the module name, *e.g.* bert.special_embeddings.0
+        module_keys = module_keys.union(set([".".join(key.split(".")[:-2]) for key in names if key[-1].isdigit()]))
```

Comment:
`torch.nn.ParameterList` is special in a sense that it adds to word with `'.'` to a module so that we need to remove the last to `"."`'s